earthfile2llb/withdockerrun.go

```diff
@@ -566,7 +414,91 @@ func makeWithDockerdWrapFun(dindID string, tarPaths []string, localRegImages []s
 	}
 }
 
-func (wdr *withDockerRun) getComposeConfig(ctx context.Context, opt WithDockerOpt) ([]byte, error) {
+func (w *withDockerRunBase) installDeps(ctx context.Context, opt WithDockerOpt) error {
+	params := composeParams(opt)
+	args := []string{
+		"/bin/sh", "-c",
+		fmt.Sprintf(
+			"%s %s",
+			strings.Join(params, " "),
+			dockerAutoInstallScriptPath),
+	}
+	runOpts := []llb.RunOption{
+		llb.AddMount(
+			dockerAutoInstallScriptPath, llb.Scratch(), llb.HostBind(), llb.SourcePath(dockerAutoInstallScriptPath)),
+		llb.Args(args),
+		llb.WithCustomNamef("%sWITH DOCKER (install deps)", w.c.vertexPrefix(false, false, false)),
+	}
+	w.c.mts.Final.MainState = w.c.mts.Final.MainState.Run(runOpts...).Root()
+	return nil
+}
+
+type withDockerRunBase struct {
+	c *Converter
+}
```

Comment:
Perhaps this should be in its own file too to help with following along.

Reply:
Good call. Updated.